Apps/AdminWebClient/test/unit/Services.Test/CommunicationService_Test.cs

```diff
@@ -0,0 +1,71 @@
+//-------------------------------------------------------------------------
+// Copyright © 2019 Province of British Columbia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-------------------------------------------------------------------------
+namespace HealthGateway.Admin.Test.Services
+{
+    using Xunit;
+    using Moq;
+    using DeepEqual.Syntax;
+    using HealthGateway.Admin.Services;
+    using HealthGateway.Database.Models;
+    using HealthGateway.Database.Wrapper;
+    using HealthGateway.Database.Delegates;
+    using Microsoft.Extensions.Logging;
+    using HealthGateway.Common.Models;
+    using System;
+    using System.Linq;
+    using HealthGateway.Admin.Models;
+    using System.Collections.Generic;
+    using HealthGateway.Common.Delegates;
+    using HealthGateway.Database.Constants;
+
+    public class CommunicationServiceTest
+    {
+        [Fact]
+        public void ShouldAddCommunication()
+        {
+            // Sample communication to test
+            Communication comm = new Communication()
+            {
+                Text = "Test communication",
+                Subject = "Testing communication",
+                EffectiveDateTime = new DateTime(2020, 04, 04),
+                ExpiryDateTime = new DateTime(2020, 05, 13)
+            };
+
+            // Set up delegate
+            DBResult<Communication> insertResult = new DBResult<Communication>
+            {
+                Payload = comm,
+                Status = DBStatusCode.Created
+            };
+
+            Mock<ICommunicationDelegate> communicationDelegateMock = new Mock<ICommunicationDelegate>();
+            communicationDelegateMock.Setup(s => s.Add(It.Is<Communication>(x => x.Text == comm.Text), true)).Returns(insertResult);
```

Comment:
Few questions, but they should not block the approval:
- are we testing also if other columns/fields will be added correctly too?
- are we testing also if text or other fields are null or empty?
- Are we expected any specific errors for these odd cases above?

Reply:
If any columns / fields are not added correctly, or are null, the assertion at the bottom of the test would fail since the data retrieved from the database would not be exactly equal to what it was given. Not sure about specific errors, the test would just fail.